sale_rental/models/stock.py

```diff
@@ -0,0 +1,242 @@
+                        'location_id': wh.rental_view_location_id.id,
+                        })
+                    slo.browse(in_loc.id).name = _('Rental In')
+                    logger.debug(
+                        'New in rental stock location created ID %d',
+                        in_loc.id)
+                wh.rental_in_location_id = in_loc.id
+            if not wh.rental_out_location_id:
+                out_loc = slo.with_context(lang='en_US').search([
+                    ('name', 'ilike', 'Rental Out'),
+                    ('location_id', '=', wh.rental_view_location_id.id),
+                    ], limit=1)
+                if not out_loc:
+                    out_loc = slo.with_context(lang='en_US').create({
+                        'name': 'Rental Out',
+                        'location_id': wh.rental_view_location_id.id,
+                        })
+                    slo.browse(out_loc.id).name = _('Rental Out')
+                    logger.debug(
+                        'New out rental stock location created ID %d',
+                        out_loc.id)
+                wh.rental_out_location_id = out_loc.id
+
+    def write(self, vals):
+        if 'rental_allowed' in vals:
+            rental_route = self.env.ref('sale_rental.route_warehouse0_rental')
+            sell_rented_route = self.env.ref(
+                'sale_rental.route_warehouse0_sell_rented_product')
+            if vals.get('rental_allowed'):
+                self._create_rental_locations()
+                self.write({
+                    'route_ids': [(4, rental_route.id)],
+                    'rental_route_id': rental_route.id,
+                    'sell_rented_product_route_id': sell_rented_route.id,
+                    })
+                for rule_vals in self._get_rental_push_pull_rules():
+                    self.env['stock.rule'].create(rule_vals)
+            else:
+                for wh in self:
+                    pull_rules_to_delete = self.env['stock.rule'].search(
+                        [
+                            ('route_id', 'in', (
+                                wh.rental_route_id.id,
+                                wh.sell_rented_product_route_id.id)),
+                            ('location_src_id', 'in', (
+                                wh.rental_out_location_id.id,
+                                wh.rental_in_location_id.id)),
+                            ('action', '=', 'move')])
+                    pull_rules_to_delete.unlink()
+                    push_rule_to_delete =\
+                        self.env['stock.rule'].search([
+                            ('route_id', '=', wh.rental_route_id.id),
+                            ('location_from_id', '=',
+                                wh.rental_out_location_id.id),
+                            ('location_dest_id', '=',
+                                wh.rental_in_location_id.id)])
+                    push_rule_to_delete.unlink()
```

Comment:
The separation of push and pull rules are not needed since they are from the same object, just a "route_id" parameter in search is enough?

If the separation is needed, then in push_rule_to_delete, the domain fields does not exists in stock.rule object and should be refactored.